ui/package.json

```diff
@@ -10,6 +10,7 @@
   "description": "UI app for Photoview",
   "scripts": {
     "start": "vite",
+    "mon": "nodemon --watch './**' --ext 'ts,json,tsx,js,html,css' --ignore './dist/' --exec 'npm start'",
```

Comment:
I guess that with the latest changes (adding the `ts-node`), the command can be simplified:
- `watch`, `ext`, and `ignore` params can be moved to the `nodeamon.json`, which would be the logical place for them, holding the complete config in 1 place
- instead of the `exec` option, it is now much better to put the path to the main TS file (as far as I understand, it is the `src/index.tsx`) - in this way we'd utilize the TypeScript JIT functionality and faster reload on changes

Reply:
I put nodemon config in `package.json`. I don't know which one is better, but in `package.json` putting all configure together.

Reply:
Please reopen this thread if you have any other ideas.

Reply:
Hm, if the config is successfully loaded from the `package.json` I'm fine with it, however, looking for it there won't be my 1st guess))

But you still use the `--exec 'npm start'` - doesn't it work with removing this parameter completely and instead of it  put the `src/index.tsx` to the nodeamon command (as in the article):
`"mon": "nodemon src/index.tsx",` 

Without this change adding TS support (looks like) doesn't add much value, because nodeamon doesn't work in the native mode.

Please let me know if you've tried that and it doesn't work for some reason.

Reply:
It's not the same way to launch a server in `photoview` as what that document says. With `nodemon src/index.tsx`, `npm run mon` fails:

```
% npm run mon

> photoview-ui@0.0.1 mon
> nodemon src/index.tsx

[nodemon] 3.1.4
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): **
[nodemon] watching extensions: ts,json,tsx,js,html,css
[nodemon] starting `node src/index.tsx`
(node:4321) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/Users/googol/code/photoview/ui/src/index.tsx:1
import React from 'react'
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (node:internal/modules/cjs/loader:1383:18)
    at Module._compile (node:internal/modules/cjs/loader:1412:20)
    at Module._extensions..js (node:internal/modules/cjs/loader:1551:10)
    at Module.load (node:internal/modules/cjs/loader:1282:32)
    at Module._load (node:internal/modules/cjs/loader:1098:12)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:215:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:158:5)
    at node:internal/main/run_main_module:30:49

Node.js v22.4.0
[nodemon] app crashed - waiting for file changes before starting...
```

Reply:
BTW, I believe there is a better way to set up `nodemon` with Typescript. I'm not a frontend expert, so I'd like to wait for others to contribute better.

Reply:
Oh, now I see...
Sure, in these circumstances we should go with the working (even not the most efficient) setup. Later, when we'll have a Frontend developer, interested in improving the setup, it will be an easy fix (I haope).
Approving the PR.